vault/token_store.go

```diff
@@ -1468,18 +1468,10 @@ func (ts *TokenStore) handleCreateCommon(
 				}
 			}
 		} else {
-			// Check against parent policies, or assign parent policies. As
-			// this is a role, add default unless explicitly disabled.
+			// Assign parent policies if none have been requested. As this is a
+			// role, add default unless explicitly disabled.
 			if len(finalPolicies) == 0 {
 				finalPolicies = policyutil.SanitizePolicies(parent.Policies, localAddDefault)
-			} else {
```

Comment:
I don't understand why this block is removed. When `allowed_roles` are not specified, not having this block will let a request for any policy to go through. No?